### PR TITLE
Add test for Matrix API Distance Calculator

### LINKED ISSUE #29: Fetch the distance matrix from the Maps Api

### PR DESCRIPTION
**Pull request by shradha-khapra**

### CODE REVIEW

File: src/test/java/com/google/sps/lib/algorithm/distance/MapsApiDistanceCalculatorTest.java [hunk 1]
```diff
@@ -34,13 +37,41 @@
...
+    double expectedDistanceMatrix[][] = {
+                                          {0.0, 1510850.0, 1404782.0},
+                                          {1509173.0, 0.0, 2053949.0},
+                                          {1409170.0, 2048291.0, 0.0},
+                                        };
+    
+    double distanceMatrix[][] = new MapsApiDistanceCalculator(mockUrlWrapper).findDistance(coordinates);
+
+    for (int i = 0; i < expectedDistanceMatrix.length; i++) {
+      assertArrayEquals(expectedDistanceMatrix[i], distanceMatrix[i], /* delta = */ 0.000001);
```
Comment (sonudoo) [inline src/test/java/com/google/sps/lib/algorithm/distance/MapsApiDistanceCalculatorTest.java]:
```suggestion
      assertArrayEquals(expectedDistanceMatrix[i], distanceMatrix[i], /* delta= */ 0.000001);
```

File: src/test/java/com/google/sps/lib/algorithm/distance/MapsApiDistanceCalculatorTest.java [hunk 2]
```diff
@@ -34,13 +37,41 @@
...
+    }
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void findDistance_failureResponse_RuntimeException_noApiResponse() throws Exception {
```
Comment (sonudoo) [inline src/test/java/com/google/sps/lib/algorithm/distance/MapsApiDistanceCalculatorTest.java]:
```suggestion
  public void findDistance_failureResponse_noApiResponse_RuntimeException() throws Exception {
```

File: src/test/java/com/google/sps/lib/algorithm/distance/MapsApiDistanceCalculatorTest.java [hunk 3]
```diff
@@ -26,6 +26,9 @@
 import java.net.HttpURLConnection;
 import org.mockito.Mock;
 import org.mockito.runners.MockitoJUnitRunner;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
 
 @RunWith(MockitoJUnitRunner.class)
 public class MapsApiDistanceCalculatorTest {
```
Comment (sonudoo) [inline src/test/java/com/google/sps/lib/algorithm/distance/MapsApiDistanceCalculatorTest.java:34]: Please add one more test case which checks that the request URL is correct for a given set of coordinates.

File: src/test/java/com/google/sps/lib/algorithm/distance/MapsApiDistanceCalculatorTest.java [hunk 4]
```diff
@@ -37,10 +40,46 @@
...
... day 2 hours\", \"value\" : 93607 }, \"status\" : \"OK\" }, { \"distance\" : { \"text\" : \"873 mi\", \"value\" : 1404782 }, \"duration\" : { \"text\" : \"23 hours 21 mins\", \"value\" : 84081 }, \"status\" : \"OK\" } ] }, { \"elements\" : [ { \"distance\" : { \"text\" : \"938 mi\", \"value\" : 1509173 }, \"duration\" : { \"text\" : \"1 day 2 hours\", \"value\" : 94436 }, \"status\" : \"OK\" }, { \"distance\" : { \"text\" : \"1 ft\", \"value\" : 0 }, \"duration\" : { \"text\" : \"1 min\", \"value\" : 0 }, \"status\" : \"OK\" }, { \"distance\" : { \"text\" : \"1,276 mi\", \"value\" : 2053949 }, \"duration\" : { \"text\" : \"1 day 17 hours\", \"value\" : 146217 }, \"status\" : \"OK\" } ] }, { \"elements\" : [ { \"distance\" : { \"text\" : \"876 mi\", \"value\" : 1409170 }, \"duration\" : { \"text\" : \"23 hours 25 mins\", \"value\" : 84308 }, \"status\" : \"OK\" }, { \"distance\" : { \"text\" : \"1,273 mi\", \"value\" : 2048291 }, \"duration\" : { \"text\" : \"1 day 16 hours\", \"value\" : 145790 }, \"status\" : \"OK\" }, { \"distance\" : { \"text\" : \"1 ft\", \"value\" : 0 }, \"duration\" : { \"text\" : \"1 min\", \"value\" : 0 }, \"status\" : \"OK\" } ] } ], \"status\" : \"OK\"}";
```
Comment (sonudoo) [inline src/test/java/com/google/sps/lib/algorithm/distance/MapsApiDistanceCalculatorTest.java:75]: Why do you need apiResponse here? Can it be empty?

**Review by sonudoo — APPROVED** [review]
LGTM!

File: src/test/java/com/google/sps/lib/algorithm/distance/MapsApiDistanceCalculatorTest.java (hunk 4 above)
Comment (shradha-khapra) [reply src/test/java/com/google/sps/lib/algorithm/distance/MapsApiDistanceCalculatorTest.java:75]: An empty response leads to an Exception when it's parsed.

**Pull request merged**